modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -118,5 +127,30 @@ def enum_zones
       logger.debug "Enumerated dns zones: #{zones}"
       zones
     end
+
+    def get_ptr_ip(zone, ip)
+      ip6 = ip.include?(':')
+      if ip6
+        zone.slice!('.ip6.arpa')
+        ip6_zone = []
+        partial_ip6 = ''
+        zone.split('.').reverse_each do |part|
+          partial_ip6 += part
+          if partial_ip6.length == 4
+            ip6_zone << partial_ip6
+            partial_ip6 = ''
+          end
+        end
+        ip6_zone + [partial_ip6] unless partial_ip6.empty?
+      end
+      if ip6
+        ip.slice!("#{ip6_zone.join(':')}:")
+        result = "#{ip.split(':').reverse.join(':')}"
+      else
+        result = "#{ip.split('.').reverse.join('.')}.in-addr.arpa"
+        result.slice!(".#{zone}")
```

Comment:
> I suspect however, you can pass in the whole address (including ip6.arpa/in-addr.arpa bit), probably ending with a '.' to add/remove record command.

This is only true in case of IPv6. IPv4 PTRs need to be zone relative; my tests adding `.` raised `DNS_ERROR_NOT_ALLOWED_UNDER_DELEGATION`